source/class/qx/tool/migration/M6_0_0.js

```diff
@@ -0,0 +1,226 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2021 The authors
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Christian Boulanger (info@bibliograph.org, @cboulanger)
+
+**********************************************************************/
+
+const process = require("process");
+const path = require("upath");
+const semver = require("semver");
+const fs = qx.tool.utils.Promisify.fs;
+
+/**
+ * Migration class for updating from v5 to v6
+ */
+qx.Class.define("qx.tool.migration.M6_0_0", {
+  extend: qx.tool.migration.BaseMigration,
+  members: {
+
+    /**
+     * Check for legacy compile.js - needs manual intervention
+     * @return {Promise<{applied: number, pending: number}>}
+     */
+    async migrateCompileJs() {
+      let migrationInfo = this.getRunner().createMigrationInfo();
+      let compileJsFilename = path.join(process.cwd(), "compile.js");
+      if (await fs.existsAsync(compileJsFilename)) {
+        let data = await fs.readFileAsync(compileJsFilename, "utf8");
```

Comment:
`existsAsync` looks unnecessary. Is there a reason not to simply try to read the file, and ignore the ENOEXIST error? (This one is likely not a race condition; just seems superfluous.)

Reply:
Sure, try/catch would work here just the same. This is just habit. But there is a drawback - if you just `try` the readFileAsync in order not to catch errors from the other code inside the `if` clause, you have to set some flag insde the try:
```javascript
let flag = false;
try { 
  let data = await fs.readFileAsync(compileJsFilename, "utf8");
  flag = true;
} catch(e){}
if (flag) { 
// do the other code that was inside the if clause
} 
 ```

I am not sure this is really superior.

Reply:
Wouldn't `finally` solve that?

Not important. Was just a thought.

Reply:
You are right! That's exactly what `finally` is for. I forgot about it.

Reply:
(I think when I first learned javascript, it didn't have finally)